src/main/java/rs/hybridit/serviceImpl/BookRentServiceImpl.java

```diff
@@ -78,6 +81,36 @@ public BookCopy getBookCopy(Book book) {
 		});
 	}
 
+	@Override
+	public List<ReportFrequency> getRentingStatistics() {
+		List<Book> books = bookRepository.findAll();
+		books.sort((a, b) -> b.getRentingCounter().compareTo(a.getRentingCounter()));
+		return books.stream().map(this::makeReport).collect(Collectors.toList());
+	}
+
+	public ReportFrequency makeReport(Book book) {
+		return new ReportFrequency(book.getName(), book.getRentingCounter());
+	}
+
+	@Override
+	public List<ReportCurrentlyRentedBooks> getCurrentlyRentedBooksReport() {
+		return bookRepository.findAll().stream()
+			.map(this::currentlyRentedBook)
+			.filter(Optional::isPresent)
+			.map(Optional::get)
+			.collect(Collectors.toList());
+	}
+
+	public Optional<ReportCurrentlyRentedBooks> currentlyRentedBook(Book book) {
+		List<BookCopy> bookCopies = bookCopyRepository.findByBook(book);
+		int available = findAvailableBookCopies(book).size();
+		int rented = bookCopies.size() - available;
+		if (rented != 0) {
```

Comment:
Isn't this just... bookCopies.size() != available :D

Reply:
It is :D